aggregator_core/src/datastore.rs

```diff
@@ -1319,7 +1319,8 @@ impl<C: Clock> Transaction<'_, C> {
                     SELECT 1 FROM client_reports
                     JOIN tasks ON tasks.id = client_reports.task_id
                     WHERE tasks.task_id = $1
-                    AND client_reports.client_timestamp <@ $2::TSRANGE
+                    AND client_reports.client_timestamp >= LOWER($2::TSRANGE)
+                    AND client_reports.client_timestamp < UPPER($2::TSRANGE)
```

Comment:
Does it make a difference if we decompose `batch_interval` into the upper and lower bounds outside of SQL?

Reply:
No, it gets constant-folded away.

`EXPLAIN ANALYZE SELECT EXISTS(SELECT 1 FROM client_reports JOIN tasks ON tasks.id = client_reports.task_id WHERE tasks.task_id = '\xc8be6ada53cb4e5d55807bcc114041f0112e9e67666fbfbb40a891c48d287135'::bytea AND client_reports.client_timestamp >= LOWER('[2024-02-07 00:00:00, 2024-02-07 00:05:00)'::tsrange) AND client_reports.client_timestamp < UPPER('[2024-02-07 00:00:00, 2024-02-07 00:05:00)'::tsrange) AND client_reports.client_timestamp >= COALESCE('2024-02-07 00:00:00'::timestamp - tasks.report_expiry_age * '1 second'::interval, '-infinity'::timestamp) AND client_reports.aggregation_started = false) AS unaggregated_report_exists;`

```
                                                                                                                                                                                                      QUERY PLAN                                                                                                                                                                                                       
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Result  (cost=8.55..8.56 rows=1 width=1) (actual time=0.026..0.027 rows=1 loops=1)
   InitPlan 1 (returns $2)
     ->  Nested Loop  (cost=0.71..24.23 rows=3 width=0) (actual time=0.024..0.025 rows=0 loops=1)
           ->  Index Scan using task_id_index on tasks  (cost=0.14..2.36 rows=1 width=16) (actual time=0.011..0.011 rows=1 loops=1)
                 Index Cond: (task_id = '\xc8be6ada53cb4e5d55807bcc114041f0112e9e67666fbfbb40a891c48d287135'::bytea)
           ->  Index Only Scan using client_reports_task_and_timestamp_unaggregated_index on client_reports  (cost=0.56..21.41 rows=46 width=16) (actual time=0.009..0.009 rows=0 loops=1)
                 Index Cond: ((task_id = tasks.id) AND (client_timestamp >= COALESCE(('2024-02-07 00:00:00'::timestamp without time zone - ((tasks.report_expiry_age)::double precision * '00:00:01'::interval)), '-infinity'::timestamp without time zone)) AND (client_timestamp >= '2024-02-07 00:00:00'::timestamp without time zone) AND (client_timestamp < '2024-02-07 00:05:00'::timestamp without time zone))
                 Heap Fetches: 0
 Planning Time: 0.436 ms
 Execution Time: 0.070 ms
(10 rows)
```